File: 20221026/모범답안/20221026_10/ControlBallNumber_test.py
import unittest

from src.com.luckydave.algorithm.fastcampus.hard.ControlBallNumber.ControlBallNumber import solution


class CustomTestCase(unittest.TestCase):

    def test_1(self):
        # given
        array = [1, 3, 4]

        # when
        result = solution(array)

        # then
        self.assertEqual(['In', 'In', 'Out', 'In', 'In', 'End'], result)

    def test_2(self):
        # given
        array = [1]

        # when
        result = solution(array)

        # then
        self.assertEqual(['In', 'End'], result)

    def test_3(self):
        # given
        array = [2]

        # when
        result = solution(array)

        # then
        self.assertEqual(['In', 'Out', 'In', 'End'], result)

    def test_4(self):
        # given
        array = []

        # when
        result = solution(array)

        # then
        self.assertEqual(['End'], result)

    def test_5(self):
        # given
        array = [2, 4]

        # when
        result = solution(array)

        # then
        self.assertEqual(['In', 'Out', 'In', 'In', 'Out', 'In', 'End'], result)

    def test_6(self):
        # given
        array = [5]

        # when
        result = solution(array)

        # then
        self.assertEqual(['In', 'Out', 'In', 'Out', 'In', 'Out', 'In', 'Out', 'In' ,'End'], result)


if __name__ == '__main__':
    unittest.main()

File: 20221026/모범답안/20221026_10/ControlBallNumber_test.js
import assert from 'assert'
import solution from './ControlBallNumber.js'

describe('Custom Test', function () {
  it('test 1', function (done) {
    // given
    const array = [1, 3, 4]

    // when
    const result = solution(array)

    // then
    assert.deepStrictEqual(['In', 'In', 'Out', 'In', 'In', 'End'], result)
    done();
  });

  it('test 2', function (done) {
    // given
    const array = [1]

    // when
    const result = solution(array)

    // then
    assert.deepStrictEqual(['In', 'End'], result)
    done();
  });

  it('test 3', function (done) {
    // given
    const array = [2]

    // when
    const result = solution(array)

    // then
    assert.deepStrictEqual(['In', 'Out', 'In', 'End'], result)
    done();
  });

  it('test 4', function (done) {
    // given
    const array = []

    // when
    const result = solution(array)

    // then
    assert.deepStrictEqual(['End'], result)
    done();
  });

  it('test 5', function (done) {
    // given
    const array = [2, 4]

    // when
    const result = solution(array)

    // then
    assert.deepStrictEqual(['In', 'Out', 'In', 'In', 'Out', 'In', 'End'], result)
    done();
  });

  it('test 6', function (done) {
    // given
    const array = [5]

    // when
    const result = solution(array)

    // then
    assert.deepStrictEqual(['In', 'Out', 'In', 'Out', 'In', 'Out', 'In', 'Out', 'In' ,'End'], result)
    done();
  });
})
